File: 1768/main.go
package main

import "fmt"

func mergeAlternately(word1 string, word2 string) string {

	word1Slice := []byte(word1)
	word2Slice := []byte(word2)
	word1Length := len(word1)
	word2Length := len(word2)
    minLength := word1Length
    maxWord := word2Slice
    result := ""

    if word1Length > word2Length {
        maxWord = word1Slice
        minLength = word2Length
    }

    k := 0

    for i := 0; i < minLength*2; i+=2{
        result += string(word1Slice[k])
        result += string(word2Slice[k])
        k++
    }
    maxWord = maxWord[k:]
    result += string(maxWord)

    return result

}

func main() {
	// result := mergeAlternately("ab", "wxyz")
	// fmt.Println(result)
	// result1 := mergeAlternately("abc", "xyz")
	// fmt.Println(result1)
	result2 := mergeAlternately("abcd", "pq")
	fmt.Println(result2)
}

File: 1768/main.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
  string mergeAlternately(string word1, string word2) {
    int w1l = word1.length();
    int w2l = word2.length();
    string maxWord = word2, minWord = word1;
    int minL = w1l;
    string result = "";
    int total = w1l + w2l;

    if (w1l > w2l) {
      minL = w2l;
      minWord = word2;
      maxWord = word1;
    }

    int k = 0;

    for (int i = 0; i < minL*2; i += 2) {
      result += word1[k];
      result += word2[k];
      cout << "W1: " << word1[k] << " W2: " << word2[k] << endl;
      k++;
      // cout << result << endl;
    }

    maxWord.erase(0, k);
    // cout << maxWord << endl;
    if(maxWord.length())
      result += maxWord;
    return result;
  }
};

int main() {
  Solution s;
  cout << s.mergeAlternately("ab", "pqrs");
}
